kbchat/kbchat_test.go

```diff
@@ -103,82 +103,290 @@ func deleteWorkingDir(workingDir string) error {
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
+}
 
 func TestSendMessageByTlfName(t *testing.T) {
-	tlfName := fmt.Sprintf("%s,%s", kbc.Username(), "kb_monbot")
-	res, err := kbc.SendMessageByTlfName(tlfName, "test")
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	text := "test SendMessageByTlfName"
+
+	// Send the message
+	res, err := alice.SendMessageByTlfName(oneOnOneChannel.Name, text)
 	require.NoError(t, err)
 	require.Greater(t, res.Result.MsgID, 0)
+
+	// Read it to confirm it sent
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	sentMessage := messages[0]
+	require.Equal(t, sentMessage.Content.Type, "text")
+	require.Equal(t, sentMessage.Content.Text.Body, text)
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
 }
 
-func TestSendMessageByTeamName(t *testing.T) {}
+func TestSendMessageByTeamName(t *testing.T) {
+	alice, _, dir, _, teamChannel := testSetup()
+	text := "test SendMessageByTeamName"
 
-func TestSendAttachmentByTeam(t *testing.T) {}
+	// Send the message
+	res, err := alice.SendMessageByTeamName(teamChannel.Name, text, &teamChannel.TopicName)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
 
-func TestReactByChannel(t *testing.T) {}
+	// Read it to confirm it sent
+	messages, err := alice.GetTextMessages(teamChannel, false)
+	require.NoError(t, err)
+	sentMessage := messages[0]
+	require.Equal(t, sentMessage.Content.Type, "text")
+	require.Equal(t, sentMessage.Content.Text.Body, text)
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
+}
+
+func TestSendAttachmentByTeam(t *testing.T) {
+	alice, _, dir, _, teamChannel := testSetup()
+	// Create a test file
+	fileName := "kb-attachment.txt"
+	location := path.Join(os.TempDir(), fileName)
+	data := []byte("My super cool attachment")
+	err := ioutil.WriteFile(location, data, 0644)
+	require.NoError(t, err)
+
+	// Send the message
+	title := "test SendAttachmentByTeam"
+	res, err := alice.SendAttachmentByTeam(teamChannel.Name, location, title, &teamChannel.TopicName)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
 
-func TestReactByConvID(t *testing.T) {}
+	// Types don't support attachments yet, so we can't read it
+	// Read it to confirm it sent
+	// messages, err := alice.GetTextMessages(teamChannel, false)
+	// require.NoError(t, err)
+	// sentMessage := messages[0]
+	// require.Equal(t, sentMessage.Content.Type, "attachment")
+	// require.Equal(t, sentMessage.Content.Attachment.Object.Title, title)
+	// require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
+}
+
+func TestReactByChannel(t *testing.T) {
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	react := ":cool:"
+	// Get last message, we'll react to it
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	lastMessageID := messages[0].MsgID
```

Comment:
if we're going to be doing something like this multiple times, id suggest pulling it out into a little helper function with a signature like `getLastMessageID(t, user, channel)`. and similar for getting the conversation id.